### DISCUSSION TITLE
ODBC Support for Metadata

### DISCUSSION THREAD

**Discussion by ev1ls33d** [juicedata/juicefs#2617] [Ideas]
Would it be feasable to add ODBC Support for Metadata? 
This way many RDBMS System can be used without explicit support.

AFAIK odbc is supported by linux and windows.

**Comment by SandyXSD** [comment]
It seems to bring more complexity, but not many benefits, since most SQL databases can be used through MySQL/PostgreSQL interfaces. We have no plan for this.

**Comment by ev1ls33d** [comment]
Aren't the most DB interfaces just referenced libs? I'm not into go but it sounds like this is <1h of work to add the ODBC dependency, when there is a go implementation ( like this one: https://github.com/alexbrainman/odbc ), I might be wrong tho.
Settings wise it could be just one complete odbc connection string like "driver=xxx;host=localhost;databaseserver=myDb;databasename=myDb;username=dba;password=pass".

**Comment by davies** [comment]
JuiceFS use xorm [1] to access SQL databases, which does not support ODBC yet. Also, the latency to database is critical, so it's better to talk to the database directly rather than using another abstract layer (proxy).

1. https://xorm.io/

**Reply by ev1ls33d** [reply]
Thanks for the information, I will read up on this.
